### PR TITLE
Get the correct content urls

### PR DESCRIPTION
**Pull request by MancunianSam**
We were originally returning the url to get the bitstream info instead
of getting the content url. This makes another API call.

It also gets the file size from the bitstream endpoint which will be
useful if we're streaming files to S3.

### CODE REVIEW

**Review by techncl — COMMENTED** [review]
Looks good; just a few comments.

File: src/main/scala/uk/gov/nationalarchives/dp/client/EntityClient.scala [hunk 1]
```diff
@@ -71,7 +71,9 @@ object EntityClient {
         allGenerationElements <- allGenerationUrls
           .map(url => getApiResponseXml(url, token))
           .sequence
-        allBitstreams <- dataProcessor.allBitstreamInfo(allGenerationElements)
+        allUrls <- dataProcessor.allBitstreamUrls(allGenerationElements)
+        bitStreams <- allUrls.map(url => getApiResponseXml(url, token)).sequence
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/dp/client/EntityClient.scala:77]: `bitstreamXmls` sounds good, I've changed it.

File: src/test/scala/uk/gov/nationalarchives/dp/client/EntityClientTest.scala [hunk 2]
```diff
@@ -98,8 +108,8 @@ abstract class EntityClientTest[F[_], S](preservicaPort: Int, secretsManagerPort
     val generationsUrl = s"/api/entity/content-objects/$ref/generations"
     val generationOneUrl = s"/api/entity/content-objects/$ref/generations/1"
     val generationTwoUrl = s"/api/entity/content-objects/$ref/generations/2"
-    val bitstreamOneUrl = s"api/entity/content-objects/$ref/generations/1/bitstreams/1"
-    val bitstreamTwoUrl = s"api/entity/content-objects/$ref/generations/2/bitstreams/2"
+    val bitstreamOneUrl = s"/api/entity/content-objects/$ref/generations/1/bitstreams/1"
+    val bitstreamTwoUrl = s"/api/entity/content-objects/$ref/generations/2/bitstreams/2"
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/dp/client/EntityClientTest.scala:112]: Ah good spot, I've changed it.

File: src/test/scala/uk/gov/nationalarchives/dp/client/EntityClientTest.scala [hunk 3]
```diff
@@ -98,8 +108,8 @@ abstract class EntityClientTest[F[_], S](preservicaPort: Int, secretsManagerPort
     val generationsUrl = s"/api/entity/content-objects/$ref/generations"
     val generationOneUrl = s"/api/entity/content-objects/$ref/generations/1"
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/dp/client/EntityClientTest.scala:109]: Yep makes sense, cheers. I've changed this one and the test above.

**Comment by MancunianSam** [comment]
I've added in a commit to refactor the Entity class as it wasn't working with circes generic decoders.

**Review by techncl — APPROVED** [review]
Looks good; thanks.

**Pull request merged**